index.js

```diff
@@ -534,13 +533,17 @@ const plugin = fp(async function (app, opts) {
     }
 
     // minJit is 0 by default
-    if (shouldCompileJit && !modifiedSchema) {
-      cached.jit = compileQuery(fastifyGraphQl.schema, modifiedDocument || document, operationName)
+    if (shouldCompileJit) {
+      if (!modifiedSchema) {
```

Comment:
```suggestion
      if (!modifiedSchema || !modifiedDocument) {
```

Reply:
If we change the `schema` but not the `document` the jit is executed against the "origin" schema.
Is it useful creating the jit when the document will be executed using the `modifiedSchema`?

Reply:
I don't understand.

> I think the modifiedDocument is worth being jit'ed it is a consistent update.
For example, a developer is renaming a field from old to new

What makes me itchy in jiting this is that the hook takes the `context` as an argument, therefore those changes could be based on user credentials. Compiling those in is a potential security hazard.

What we really need is another field to defer this to the user. Having a `cacheable` or `jit` property would put the developer in charge of what is happening.

Reply:
I get the point, I will try to arrange a security test for this case.

The current PR's code blocks the jit execution when the schema is modified by the hook.
The suggested code enables the jit when the user changes the `modifiedSchema` property that seems unwanted.

> What we really need is another field to defer this to the user. Having a cacheable or jit property would put the developer in charge of what is happening.

Clear, I would add it in a follow-up PR because there are a few cases that must be discussed or would you like to upgrade this PR instead?

The doubts concern:

- should we provide the cache counter to the hook function using the context?
- does the `cacheable` property have priority over the `jit` option?

Reply:
> should we provide the cache counter to the hook function using the context?

Unsure about this

> does the cacheable property have priority over the jit option?

Yes